src/cmd/flash.js

```diff
@@ -55,11 +62,136 @@ module.exports = class FlashCommand extends CLICommandBase {
 		} else if (local) {
 			let allFiles = binary ? [binary, ...files] : files;
 			await this.flashLocal({ files: allFiles, applicationOnly, target });
+		} else if (tachyon) {
+			let allFiles = binary ? [binary, ...files] : files;
+			await this.flashTachyon({ verbose, files: allFiles });
 		} else {
 			await this.flashCloud({ device, files, target });
 		}
 	}
 
+	async flashTachyon({ verbose, files }) {
+		this.ui.write(`${os.EOL}Ensure only one device is connected to the computer${os.EOL}`);
+
+		let zipFile;
+		let includeDir = '';
+		let updateFolder = '';
+
+		if (files.length === 0) {
+			// If no files are passed, use the current directory
+			files = ['.'];
+		}
+
+		const input = files[0];
+		const stats = await fs.stat(input);
+		let filesToProgram;
+
+		if (stats.isDirectory()) {
+			updateFolder = path.dirname(input);
+			const dirInfo = await this._extractFlashFilesFromDir(input);
+			includeDir = path.join(path.basename(input), dirInfo.baseDir);
+			filesToProgram = dirInfo.filesToProgram;
+		} else if (utilities.getFilenameExt(input) === '.zip') {
+			updateFolder = path.dirname(input);
+			zipFile = path.basename(input);
+			const zipInfo = await this._extractFlashFilesFromZip(input);
+			includeDir = path.normalize(zipInfo.baseDir);
```

Comment:
`includeDir = path.normalize(zipInfo.baseDir);`

This gives includeDir as `--include abc/images/qcmxxxx` but the qdl (atleast on mac) needs to have `--include abc/abc/images/qcmxxxx`

This is the folder structure
```
abc.zip
|
|-- abc
|   |--images
|   |     |--qcmxxx
|   |     |   |--prog_firehose_ddr.elf
|   |     |.  |--rawprogram0.elf
```